Add possibility to choose desired arrival time

Elhebert opened the pull request:
Hi,

I would like to be able to know when I have to leave to be home around a given time. With the implementation of the api-key done in 86d4687 and 031622b, you can use transit mode, and choose an `arrival_time` instead of the `departure_time`.

The downside is, it's only available in transit mode, the `arrival_time` paramater is not supported in other mode.

Use: 

```
User> hubot i wanna be home by 16h00
Hubot> From X to Y, you'll have to leave at 15h30
```

Commit:
Add possibility to choose desired arrival time

Commit:
Update README.me

Cinetik commented on src/traffic-status.coffee:
```diff
@@ -48,3 +48,32 @@ module.exports = (robot) ->
+		robot.respond /i wanna be home by (\d\d)(?:h|:)(\d\d)/i, (msg) ->
+		@userbrain = robot.brain.userForName(msg.message.user.name)
+		if not @userbrain.mode
+			@userbrain.mode = "transit"
```
Why transit by default while it is driving on the other function ?

Commit:
Fix indent error

Cinetik commented on src/traffic-status.coffee:
```diff
@@ -48,3 +48,33 @@ module.exports = (robot) ->
+	robot.respond /i wanna be home by (\d\d)(?:h|:)(\d\d)/i, (msg) ->
+		@userbrain = robot.brain.userForName(msg.message.user.name)
+		if not @userbrain.mode
+			@userbrain.mode = "transit"
```
Why transit by default while it is driving on the other function ?

Elhebert commented:
My bad, misread a part of the documentation. I'll fix it.

Commit:
Support of the 12h format (am/pm)

Commit:
Update README.md

Elhebert commented:
I corrected the default mode error.

I also added the possibility to set the arrival time in am/pm or in 24h format. Hubot will also now answer the user with the time based on the local set on the server.
An other option that we could implement is to answer with the format the user made his request. If the `arrival_time` is set with am/pm, hubot answer in amp/pm, in 24h format otherwise.
